### ISSUE TITLE
dis_loss and div_loss do not decrease

### ISSUE THREAD

**Issue by kli017** [Jianlong-Fu/Multi-Attention-CNN#9]
Hello,
I was trying to train the channel grouping network on CUB dataset. However I found that the div_loss and dis_loss do decrease from the very beginning(fixing on 22 and 43). I was fine-tune using the vgg16 weights on imagenet and fixed the parameters of convolution layer at first according to the paper. Could you give me some suggestion to train the model properly?
![image](https://user-images.githubusercontent.com/14877573/44828171-36655200-ac49-11e8-92b9-4c78657bd1ea.png)

**Comment by Blue-Clean** [comment]
Hello. I am new to Caffe. May I ask you how to train the MA-CNN? How to get the train.prototxt?(Only add the loss layer in deploy.prototxt)? And how to start training process and get the .caffemode(edit a  *.py file using pycaffe or using batch command such as solver.prototxt using caffe )?

**Comment by kli017** [comment]
@Blue-Clean Hello, if you are using Baidu Cloud, Someone shared the link in #2 . You can find train.prototxt there.  To train the MA-CNN, according to my understand, you need to train the part_model first to crop attention part first and then train the classification model. I was doing the first part however the result was not good. I don not know if there any special trick to train the model.

**Comment by Blue-Clean** [comment]
@kli017 Thanks for your reply.  i have already downloaded the release file.Only bird_part_deploy.prototxt and bird_cls_deploy.prototxt are shared but not train.prototxt. So. I need to create train.prototxt based on deploy.prototxt(add the loss layer in shared deploy.prototxt)?

**Comment by wuyongchn** [comment]
can you share the DisLoss and DivLoss code?

**Comment by yungng-lab** [comment]
can you share the DisLoss and DivLoss code? @kli017

**Comment by Blue-Clean** [comment]
@kli017 Hello,can you tell me how to train the part_cnn ? I  get into trouble to define DisLoss and DivLoss .

**Comment by yungng-lab** [comment]
@Blue-Clean Hello, do you get the DisLoss and DivLoss code from #2 ?

**Comment by kli017** [comment]
@Blue-Clean @wuyongchn hello I get the prototxt file from this link https://github.com/minfengUCAS/Muti-Attention-CNN

**Comment by yungng-lab** [comment]
@kli017 thank you very much

**Comment by Zxlan** [comment]
@kli017 Hello,can you tell me how to add the transpose layer?I cannot know how to refer https://github.com/houkai/caffe.

**Comment by 2501231120** [comment]
@Blue-Clean .Hello, I want to ask you how I can write train program if I want to train my own data set.

**Comment by 2501231120** [comment]
Hello, I want to ask you how I can write train program if I want to train my own data set.

**Comment by zsano1** [comment]
Hello guys, after I clone the caffe which contains "transform" layer, and `cmake`, `make`. But the error still says it doesn't contain "transform". I guess I may need to import this newly built caffe and` sys.append("..")`. But it seems not work. Can you by any chance give some suggestions? @Blue-Clean @kli017 @wuyongchn
